internal/storage/sql/mysql/mysql.go

```diff
@@ -147,6 +147,25 @@ func (s *Store) CreateConstraint(ctx context.Context, r *flipt.CreateConstraintR
 	return constraint, nil
 }
 
+func (s *Store) CreateRollout(ctx context.Context, r *flipt.CreateRolloutRequest) (*flipt.Rollout, error) {
+	rollout, err := s.Store.CreateRollout(ctx, r)
+
+	if err != nil {
+		var merr *mysql.MySQLError
+
+		if errors.As(err, &merr) && merr.Number == constraintForeignKeyErrCode {
+			if segment := r.GetSegment(); segment != nil {
+				return nil, errs.ErrNotFoundf(`flag "%s/%s or segment %s"`, r.NamespaceKey, r.FlagKey, segment.SegmentKey)
+			}
+			return nil, errs.ErrNotFoundf(`flag "%s/%s"`, r.NamespaceKey, r.FlagKey)
```

Comment:
Same as with postgres, I think we should poke around in the `*mysql.MySQLError` a bit and see if we can figure out how to become more determistic here with what error is returned.